rfcs/0004-leader-rotation.md

```diff
@@ -1,29 +1,23 @@
 # Leader Rotation
 
-The goal of this RFC is to define how leaders rotate in taking their role.
+The goal of this RFC is to define how fullnodes rotate in taking the leader role.
```

Comment:
Leaders is more accurate.  A stake could configure a leader, maybe multiple leaders with a different weight.

Reply:
There's no such thing as "leaders" in the codebase. Staked fullnodes rotate taking the leader role. All other fullnodes take a validator role. The validators aren't even ranked to take the leader's role in its absence. Instead, the next leader in the rotation builds on empty entries (what you've been calling virtual ticks).

Reply:
Not every fullnode is in rotation, just the ones that are assigned a ‘leader’ function by a stake.

Reply:
Assigned a leader function by whom? I don't think I've ever seen that written down, and if it were, I'd expect it to be in this document. Can you point me to a more detailed description or post it here?

Reply:
So you're saying that when a fullnode stakes, it should opt in as a potential leader? Does that imply it's staked as a validator as well (when not a leader)?

Reply:
@garious Separate identities for separate functions.  A staking program state has an assignment for leader, validator, etc.., as the identities for those functions. We basically want to limit the exposure of leaking a key to the functions that it serves and only those.